src/app_service/service/community/service.nim

```diff
@@ -1351,6 +1356,59 @@ QtObject:
       self.events.emit(SIGNAL_COMMUNITY_IMPORTED, CommunityArgs(community: community))
 
     self.events.emit(SIGNAL_COMMUNITY_DATA_IMPORTED, CommunityArgs(community: community))
+  
+  proc removeLoadingRequestToJoinCommunity(self: Service, communityId: string) =
+    var i = 0
+    for communityRequest in self.myCommunityRequests:
+      if (communityRequest.communityId == communityId and RequestToJoinType(communityRequest.state) == RequestToJoinType.Loading):
+          self.myCommunityRequests.delete(i)
+          self.events.emit(SIGNAL_REQUEST_TO_JOIN_COMMUNITY_DELETED, CommunityRequestToJoinArgs(communityId: communityId))
+          return
+      i.inc()
+
+  proc asyncRequestToJoinCommunity*(self: Service, communityId: string, ensName: string, password: string) =
+    try:
+      let communityRequest = CommunityMembershipRequestDto(
+        id: "",
+        publicKey: "",
+        chatId: "",
+        communityId: communityId,
+        state: int(RequestToJoinType.Loading),
+        our: ""
+      )
+      self.myCommunityRequests.add(communityRequest)
+      self.events.emit(SIGNAL_COMMUNITY_MY_REQUEST_ADDED, CommunityRequestArgs(communityRequest: communityRequest))
+
+      let arg = AsyncRequestToJoinCommunityTaskArg(
+        tptr: cast[ByteAddress](asyncRequestToJoinCommunityTask),
+        vptr: cast[ByteAddress](self.vptr),
+        slot: "onAsyncRequestToJoinCommunityDone",
+        communityId: communityId,
+        ensName: ensName,
+        password: password
+      )
+      self.threadpool.start(arg)
+    except Exception as e:
+      self.removeLoadingRequestToJoinCommunity(communityId)
+      error "Error request to join community", msg = e.msg 
+    
+  proc onAsyncRequestToJoinCommunityDone*(self: Service, communityIdAndRpcResponse: string) {.slot.} =
+    try:
+      let rpcResponseObj = communityIdAndRpcResponse.parseJson
+      if (rpcResponseObj{"response"}{"error"}.kind != JNull):
+        let error = Json.decode($rpcResponseObj["response"]["error"], RpcError)
+        error "Error requesting community info", msg = error.message
+        return
+
+      let communityId = rpcResponseObj{"communityId"}.getStr()
+      let rpcResponse = Json.decode($rpcResponseObj["response"], RpcResponse[JsonNode])
+      self.activityCenterService.parseActivityCenterResponse(rpcResponse)
+      
+      if not self.processRequestsToJoinCommunity(rpcResponse.result):
+        error "error: ", procName="onAsyncRequestToJoinCommunityDone", errDesription = "no 'requestsToJoinCommunity' key in response"
+
```

Comment:
Here's where I think we should emit `ADDED` event (or `UPDATED`, but probably `ADDED` because we're only adding membership requests, we're never really updating an existing one